### PR TITLE
Adding recipes for Adaflow package

### PR DESCRIPTION
**Pull request by RobinQu**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by RobinQu** [comment]
@conda-forge/help-c-cpp  Please review this receipe.

File: recipes/adaflow/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+    - gst-plugins-ugly  1.22.0
+    - gst-python  1.22.0
+  run:
+    - {{ pin_compatible("glib") }}
+
+
+test:
+  commands:
+    - "test -f $PREFIX/lib/libflowmetadata.*"
+    - "test -f $PREFIX/lib/gstreamer-1.0/libflow_gst_plugin.*"
```
Comment (loriab) [inline recipes/adaflow/meta.yaml:70]:
```suggestion
    - "test -f $PREFIX/lib/libflowmetadata$SHLIB_EXT"
    - "test -f $PREFIX/lib/gstreamer-1.0/libflow_gst_plugin$SHLIB_EXT"
```
right? the var will supply `.so` on linux and `.dylib` on mac

File: recipes/adaflow/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+
+# test:
```
Comment (loriab) [inline recipes/adaflow/meta.yaml:72]: It's strongly recommended to test an import, or a help command, or some small execution to confirm in working order and the requirements/run is populated.

Comment (loriab) [reply recipes/adaflow/meta.yaml]: Thanks for the new test. Are these "failed to load plugin" messages expected? https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=689896&view=logs&j=6f142865-96c3-535c-b7ea-873d86b887bd&t=22b0682d-ab9e-55d7-9c79-49f3c3ba4823&l=2260

Comment (RobinQu) [reply recipes/adaflow/meta.yaml]: Some gst plugins like `libgstopengl.so` and `libgstnvcodec.so`  are installed but cannot be loaded by `gst-inpect-1.0` due to missing devices, system dependencies.  It's more related GStreamer itself and it's expected in this build env.

Comment (RobinQu) [reply recipes/adaflow/meta.yaml]: > Thanks for the new test. Are these "failed to load plugin" messages expected? https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=689896&view=logs&j=6f142865-96c3-535c-b7ea-873d86b887bd&t=22b0682d-ab9e-55d7-9c79-49f3c3ba4823&l=2260

Some gst plugins like `libgstopengl.so` and `libgstnvcodec.so`  are installed but cannot be loaded by `gst-inpect-1.0` due to missing devices, system dependencies.  It's more related GStreamer itself and it's expected in this build env.

File: recipes/adaflow/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,96 @@
+# Note: there are many handy hints in comments in this example -- remove them when you've finalized your recipe
+# If your package is python based, we recommend using Grayskull to generate it instead:
+  # https://github.com/conda-incubator/grayskull
+
+  # Jinja variables help maintain the recipe as you'll update the version only here.
+  # Using the name variable with the URL in line 14 is convenient
+  # when copying and pasting from another recipe, but not really needed.
+  {% set name = "adaflow" %}
+  {% set version = "v0.0.1" %}
```
[the remark this answers is not in the document]
Comment (RobinQu) [reply recipes/adaflow/meta.yaml]: version number in CMakeLists.txt is aligned to v0.0.1.

File: recipes/adaflow/build.sh [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+set -x
+
+#rm -rf build && mkdir build && cd build && LD_LIBRARY_PATH=$CONDA_PREFIX/lib:$LD_LIBRARY_PATH \
+#PKG_CONFIG_PATH=$CONDA_PREFIX/lib/pkgconfig:$PKG_CONFIG_PATH \
+#LDFLAGS="$LDFLAGS -L$CONDA_PREFIX/lib -Wl,-rpath,$CONDA_PREFIX/lib" \
+ls -la
+echo $CC
+echo $CXX
+mkdir build && cd build
+cmake \
```
[the remark this answers is not in the document]
Comment (RobinQu) [reply recipes/adaflow/build.sh:8]: CMakeLists.txt now responde to `CMAKE_CXX_FLAGS` and `CMAKE_CXX_FLAGS_DEBUG` options. And `${CMAKE_ARGS}` is added to script.

**Comment by RobinQu** [comment]
@loriab Is there any way to speed up this PR?

**Comment by RobinQu** [comment]
@conda-forge/staged-recipes @conda-forge/help-python-c ready for review now.

**Comment by RobinQu** [comment]
@conda-forge-admin please ping @conda-forge/staged-recipes @conda-forge/help-python-c to review.

File: recipes/adaflow/build.sh [hunk 5]
```diff
@@ -0,0 +1,14 @@
...
+set -x
+
+mkdir build && cd build
+cmake \
+      ${CMAKE_ARGS} \
+      -DPKG_CONFIG_EXECUTABLE=$CONDA_PREFIX/bin/pkg-config \
+      -DCMAKE_BUILD_TYPE=Release \
+      -DCMAKE_INSTALL_PREFIX=$PREFIX ..
+make -j${nproc} && \
+make install
```
Comment (ocefpaf) [inline recipes/adaflow/build.sh:14]: Is there a make check/test or ctest we can run there?

Comment (RobinQu) [reply recipes/adaflow/build.sh:14]: To validate installation itself, `gst-inspect-1.0 flow_gst_plugin` wil be enough, as GST will acutally attempt to load the library and report.

Integration tests are written in python, and it requires extra [pyhon wheel](https://pypi.org/project/adaflow-python/) to run.

**Comment by RobinQu** [comment]
@ocefpaf I made a mistake by creating a `LICENSE` file at source root. I thought it was neccessary for receipe to ship with its own license file. And I find out it's merged to master. 

Should I create a reverse commit to recover original license file?

**Pull request merged**